WPFSKillTree/ViewModels/PassiveTree/PassiveNodeViewModel.cs

```diff
@@ -0,0 +1,147 @@
+﻿using PoESkillTree.Engine.GameModel;
+using PoESkillTree.Engine.GameModel.PassiveTree;
+using PoESkillTree.Engine.GameModel.PassiveTree.Base;
+using PoESkillTree.Engine.Utils.Extensions;
+using PoESkillTree.SkillTreeFiles;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace PoESkillTree.ViewModels.PassiveTree
+{
+    public class PassiveNodeViewModel
+    {
+        private readonly JsonPassiveNode JsonPassiveNode;
+        public readonly PassiveNodeGroupViewModel? PassiveNodeGroup;
+
+        public PassiveNodeViewModel() : this(new JsonPassiveNode()) { }
+        public PassiveNodeViewModel(JsonPassiveNode jsonPassiveNode, PassiveNodeGroupViewModel? group = null)
+        {
+            JsonPassiveNode = jsonPassiveNode;
+            PassiveNodeGroup = group;
+            InitializeAttributes();
+        }
+        public PassiveNodeDefinition PassiveNodeDefinition => PassiveNodeDefinition.Convert(JsonPassiveNode);
+        public CharacterClass? StartingCharacterClass { get => JsonPassiveNode.StartingCharacterClass; set => JsonPassiveNode.StartingCharacterClass = value; }
+        public string[]? Recipe { get => JsonPassiveNode.Recipe; set => JsonPassiveNode.Recipe = value; }
+        public JsonExpansionJewelSocket? ExpansionJewelSocket { get => JsonPassiveNode.ExpansionJewelSocket; set => JsonPassiveNode.ExpansionJewelSocket = value; }
+        public HashSet<ushort> OutPassiveNodeIds => JsonPassiveNode.OutPassiveNodeIds;
+        public HashSet<ushort> InPassiveNodeIds => JsonPassiveNode.InPassiveNodeIds;
+        public bool IsSkilled { get => JsonPassiveNode.IsSkilled; set => JsonPassiveNode.IsSkilled = value; }
```

Comment:
There are multiple unused properties in here that I don't really mind, but this one doesn't really make sense for what this class is used for and might even be misleading.

Reply:
ExpansionJewelSocket will be needed if we plan to support that piece. I have removed the IsSkilled property